pdb2pqr/biomolecule.py

```diff
@@ -265,21 +268,24 @@ def set_termini(self, neutraln=False, neutralc=False):
                     ) and not residue.is3term:
                         fixflag = 1
                 if fixflag:
-                    # Get an available chain ID
-                    chainid = letters[0]
-                    id_ = 0
-                    id_length = 1
-                    while chainid in self.chainmap:
-                        id_ += 1
-                        if id_ >= len(letters):
-                            id_length += 1
-                            id_ = 0
-                        chainid = letters[id_] * id_length
-                    if id_length > 1:
-                        message = (
-                            "Warning: Reusing chain id: " + chainid[0] + ""
-                        )
-                        _LOGGER.warning(message)
+                    chainid = residue.chain_id
+                    if chainid in seen_chains:
```

Comment:
I don't understand the logic here -- should this be `if chainid not in seen_chains`?  Can you please add some comments explaining?

Reply:
If the chain name has not been seen before (no such chain was added in the past) don't generate a new name.
If the chain name has been seen before (already added such a chain) generate a new name.

I will try to add some tests to this PR as well though

Reply:
@stefdoerr do you have a test case that exposes the problem before the code change. Also, @sobolevnrm did you want Stefan's comments on this PR or some comments in the actual code? I would prefer the comments be put in the code.

Reply:
The actual code, if possible.